src/MESONT/angle_mesocnt.cpp

```diff
@@ -0,0 +1,392 @@
+      a = kb[type] * s;
+
+      buckled[i2] = 1;
+    }
+    a11 = a * c / rsq1;
+    a12 = -a / (r1 * r2);
+    a22 = a * c / rsq2;
+
+    f1[0] = a11 * delx1 + a12 * delx2;
+    f1[1] = a11 * dely1 + a12 * dely2;
+    f1[2] = a11 * delz1 + a12 * delz2;
+    f3[0] = a22 * delx2 + a12 * delx1;
+    f3[1] = a22 * dely2 + a12 * dely1;
+    f3[2] = a22 * delz2 + a12 * delz1;
+
+    // apply force to each of 3 atoms
+
+    if (newton_bond || i1 < nlocal) {
+      f[i1][0] += f1[0];
+      f[i1][1] += f1[1];
+      f[i1][2] += f1[2];
+    }
+
+    if (newton_bond || i2 < nlocal) {
+      f[i2][0] -= f1[0] + f3[0];
+      f[i2][1] -= f1[1] + f3[1];
+      f[i2][2] -= f1[2] + f3[2];
+    }
+
+    if (newton_bond || i3 < nlocal) {
+      f[i3][0] += f3[0];
+      f[i3][1] += f3[1];
+      f[i3][2] += f3[2];
+    }
+
+    if (evflag)
+      ev_tally(i1, i2, i3, nlocal, newton_bond, eangle, f1, f3, delx1, dely1, delz1, delx2, dely2,
+               delz2);
+  }
+}
+
+/* ---------------------------------------------------------------------- */
+
+void AngleMesoCNT::allocate()
+{
+  allocated = 1;
+  const int np1 = atom->nangletypes + 1;
+
+  memory->create(buckling, np1, "angle:buckling");
+  memory->create(kh, np1, "angle:kh");
+  memory->create(kb, np1, "angle:kb");
+  memory->create(thetab, np1, "angle:thetab");
+  memory->create(theta0, np1, "angle:theta0");
+
+  memory->create(setflag, np1, "angle:setflag");
+  for (int i = 1; i < np1; i++) setflag[i] = 0;
+}
+
+/* ----------------------------------------------------------------------
+   set coeffs for one or more types
+------------------------------------------------------------------------- */
+
+void AngleMesoCNT::coeff(int narg, char **arg)
+{
+  if (narg < 1) error->all(FLERR, "Incorrect args for angle coefficients");
+
+  int buckling_one;
+  if (strcmp(arg[1], "buckling") == 0)
+    buckling_one = 1;
+  else if (strcmp(arg[1], "harmonic") == 0)
+    buckling_one = 0;
+  else
+    error->all(FLERR,
+               "Unknown first argument for angle coefficients, must be 'buckling' or 'harmonic'");
+
+  // units, eV to energy unit conversion
+
+  double ang = force->angstrom;
+  double eunit;
+  if (strcmp(update->unit_style, "lj") == 0)
+    error->all(FLERR, "Angle style mesocnt does not support lj units");
+  else if (strcmp(update->unit_style, "real") == 0)
+    eunit = 23.06054966;
+  else if (strcmp(update->unit_style, "metal") == 0)
+    eunit = 1.0;
+  else if (strcmp(update->unit_style, "si") == 0)
+    eunit = 1.6021765e-19;
+  else if (strcmp(update->unit_style, "cgs") == 0)
+    eunit = 1.6021765e-12;
+  else if (strcmp(update->unit_style, "electron") == 0)
+    eunit = 3.674932248e-2;
+  else if (strcmp(update->unit_style, "micro") == 0)
+    eunit = 1.6021765e-4;
+  else if (strcmp(update->unit_style, "nano") == 0)
+    eunit = 1.6021765e2;
+  else
+    error->all(FLERR, "Angle style mesocnt does not recognize this units style");
```

Comment:
```suggestion
    error->all(FLERR, "Angle style mesocnt does not support {} units", update->unit_style);
```

With this the first if statement could be removed, too, since this line will provide the same error message.